src/test/java/pages/AllSongsPage.java

```diff
@@ -1,45 +1,72 @@
-    By searchField = By.cssSelector("[name='q']");
+    @FindBy(xpath = "//div[contains(text(), 'new playlist17')]")
+    WebElement successNewPlaylist;
+
+    @FindBy(xpath = "//*[@id='songResultsWrapper']//button[@title='Save']")
+    WebElement savePlaylistButton;
+
+    @FindBy(xpath = "//*[@id='songResultsWrapper']//input[@data-test='new-playlist-name']")
+    WebElement createPlaylistField;
+
+    @FindBy(css = "[class='btn-add-to']")
+    WebElement addToButton;
+
+    @FindBy(xpath = "(//*[@id='songResultsWrapper']//tr[contains(@class,'song-item')])[1]")
+    WebElement firstSong;
+
+    @FindBy(css = "[data-test='view-all-songs-btn']")
+    WebElement viewAllButton;
+
+    @FindBy(css = "[name='q']")
+    WebElement searchField;
+//    By successNewPlaylist = By.xpath("//div[contains(text(), 'new playlist17')]");
+//    By savePlaylistButton = By.xpath("//*[@id='songResultsWrapper']//button[@title='Save']");
+//    By createPlaylistField = By.xpath("//*[@id='songResultsWrapper']//input[@data-test='new-playlist-name']");
+//    By addToButton = By.cssSelector("[class='btn-add-to']");
+//    By firstSong = By.xpath("(//*[@id='songResultsWrapper']//tr[contains(@class,'song-item')])[1]");
+//    By viewAllButton = By.cssSelector("[data-test='view-all-songs-btn']");
+//    By searchField = By.cssSelector("[name='q']");
 
 
 
     public WebElement verifySuccessMessageText() {
-        return findElement(successNewPlaylist);
+        return successNewPlaylist;
     }
-    public void selectSavePlaylist() {
-        findElement(savePlaylistButton).click();
+    public AllSongsPage selectSavePlaylist() {
+        savePlaylistButton.click();
+        return this;
     }
-    public void enterNewPlaylistName(String newPlaylistName) {
-        findElement(createPlaylistField).click();
-        findElement(createPlaylistField).sendKeys(newPlaylistName);
+    public AllSongsPage enterNewPlaylistName(String newPlaylistName) {
+        createPlaylistField.click();
+        createPlaylistField.sendKeys(newPlaylistName);
+        return this;
     }
-    public void clickAddTo() {
-        findElement(addToButton).click();
+    public AllSongsPage clickAddTo() {
+        addToButton.click();
+        return this;
     }
-    public void clickFirstSong() {
-        findElement(firstSong).click();
+    public AllSongsPage clickFirstSong() {
+        firstSong.click();
+        return this;
     }
-    public void clickViewAll() {
-        findElement(viewAllButton).click();
+    public AllSongsPage clickViewAll() {
+        viewAllButton.click();
+        return this;
     }
-    public void searchSongName(String songName) {
-        findElement(searchField).click();
-        findElement(searchField).sendKeys(songName);
+    public AllSongsPage searchSongName(String songName) {
+        //wait.until(ExpectedConditions.elementToBeClickable(searchField)).click();
+        searchField.click();
+        searchField.sendKeys(songName);
+        return this;
```

Comment:
The function may be more appropriate to write them directly inside the method. This keeps the code self-contained and easier to understand. In some cases, the returned value from the method fails to function correctly.
![Screenshot_29](https://github.com/Bogin404/A47-QA-Automation-Framework/assets/25093177/2c998e88-5823-4e51-a7e8-5d3d87279aa2)
![Screenshot_30](https://github.com/Bogin404/A47-QA-Automation-Framework/assets/25093177/c4ef3cf3-71c1-45b1-b65c-823d31ac43c9)